kimchi/src/circuits/constant_polynomial.rs

```diff
@@ -0,0 +1,71 @@
+use crate::circuits::constraints::ZK_ROWS;
+use crate::circuits::domains::EvaluationDomains;
+use ark_ff::FftField;
+use ark_poly::UVPolynomial;
+use ark_poly::{univariate::DensePolynomial as DP, Evaluations as E, Radix2EvaluationDomain as D};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+#[serde_as]
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct ConstantPolynomial<F: FftField> {
+    /// 1-st Lagrange evaluated over domain.d8
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
+    pub l1: E<F, D<F>>,
+    /// 0-th Lagrange evaluated over domain.d4
+    // TODO(mimoo): be consistent with the paper/spec, call it L1 here or call it L0 there
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
+    pub l04: E<F, D<F>>,
+    /// 0-th Lagrange evaluated over domain.d8
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
+    pub l08: E<F, D<F>>,
+    /// zero evaluated over domain.d8
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
+    pub zero4: E<F, D<F>>,
+    /// zero evaluated over domain.d8
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
+    pub zero8: E<F, D<F>>,
```

Comment:
I think zero4 and zero8 can be deleted, I don't see them used (at least in this PR, not sure if tests pass)

Reply:
i think you are right. planning to look into that next